beacon_node/store/src/hot_cold_store.rs

```diff
@@ -777,7 +788,22 @@ impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> HotColdDB<E, Hot, Cold>
         )?;
 
         // 3. Replay the blocks on top of the low restore point.
-        self.replay_blocks(low_restore_point, blocks, slot, BlockReplay::Accurate)
+        // Use a forwards state root iterator to avoid doing any tree hashing.
+        // The state root of the high restore point should never be used, so is safely set to 0.
```

Comment:
Do you think there's any value in something like this:

```rust
let high_restore_point_root = if high_restore_point.slot() < slot {
  Hash256::zero()
} else {
  return Err(...)
};
```

Reply:
I think this particular check would be overdoing it a bit because lots of other things would break really badly if the high restore point's slot wasn't greater than `slot`. e.g. we'd try to replay blocks with bad slots, which would error in block processing.